src/drivers/wdm/libraries/WdmDrivers.qll

```diff
@@ -80,24 +81,137 @@ class WdmCallbackRoutine extends Function {
   }
 }
 
+
+/** A WDM DriverEntry callback routine. */
+class WdmDriverEntry extends WdmCallbackRoutine {
+  WdmDriverEntry() { callbackType.getName().matches("DRIVER_INITIALIZE") }
+
+  string getExpectedMaxIrqlLevelString() { result = "PASSIVE_LEVEL" }
+
+  string getExpectedMinIrqlLevelString() { result = "PASSIVE_LEVEL " }
+}
+
+/** A WDM DrierStartIo callback routine */
+class WdmDriverStartIo extends WdmCallbackRoutine {
+  WdmDriverStartIo() { callbackType.getName().matches("DRIVER_STARTIO") }
+
+  string getExpectedMaxIrqlLevelString() { result = "DISPATCH_LEVEL" }
+
+  string getExpectedMinIrqlLevelString() { result = "PASSIVE_LEVEL " }
+}
+
+/**
+ * A WDM DriverUnload callback routine.
+ */
+class WdmDriverUnload extends WdmCallbackRoutine {
+  WdmDriverUnload() { callbackType.getName().matches("DRIVER_UNLOAD") }
+
+  string getExpectedMaxIrqlLevelString() { result = "PASSIVE_LEVEL" }
+
+  string getExpectedMinIrqlLevelString() { result = "PASSIVE_LEVEL " }
+}
+
+// NOTE duplicate for backward compatibility with other query. Remove when other query is updated.
 /** A WDM AddDevice callback routine. */
 class WdmAddDevice extends WdmCallbackRoutine {
   WdmAddDevice() { callbackType.getName().matches("DRIVER_ADD_DEVICE") }
 }
 
-/** A WDM DriverUnload callback routine. */
-class WdmDriverUnload extends WdmCallbackRoutine {
-  WdmDriverUnload() { callbackType.getName().matches("DRIVER_UNLOAD") }
+/**
+ * A WDM DriverAddDevice callback routine.
+ */
+class WdmDriverAddDevice extends WdmCallbackRoutine {
+  WdmDriverAddDevice() { callbackType.getName().matches("DRIVER_ADD_DEVICE") }
+
+  string getExpectedMaxIrqlLevelString() { result = "PASSIVE_LEVEL" }
+
+  string getExpectedMinIrqlLevelString() { result = "PASSIVE_LEVEL " }
 }
 
-/** A WDM DriverEntry callback routine. */
-class WdmDriverEntry extends WdmCallbackRoutine {
-  WdmDriverEntry() { callbackType.getName().matches("DRIVER_INITIALIZE") }
+/**
+ * A WDM DriverDispatch callback routine.
+ */
+class WdmDriverDispatch extends WdmCallbackRoutine {
+  WdmDriverDispatch() { callbackType.getName().matches("DRIVER_DISPATCH") }
+
+  string getExpectedMaxIrqlLevelString() { result = "PASSIVE_LEVEL" }
+
+  string getExpectedMinIrqlLevelString() { result = "PASSIVE_LEVEL " }
 }
 
-/** A WDM DriverCancel callback routine. */
+/**
+ * A WDM IO completion routine.
+ */
+class WdmDriverCompletionRoutine extends WdmCallbackRoutine {
+  WdmDriverCompletionRoutine() { callbackType.getName().matches("IO_COMPLETION_ROUTINE") }
+
+  string getExpectedMaxIrqlLevelString() { result = "DISPATCH_LEVEL " }
```

Comment:
Quite a few of these IRQL strings have a space inside them at the end.  Please remove the spaces.